local_planner/src/local_planner.cpp

```diff
@@ -255,16 +256,24 @@ void LocalPlanner::getStateAndReferencePlan() {
   // Grab the appropriate states from the body plan and convert to an Eigen
   // matrix
   ref_body_plan_.setZero();
+  ref_primitive_plan_.setZero();
 
   for (int i = 0; i < N_; i++) {
     // If the horizon extends past the reference trajectory, just hold the last
     // state
     if (i + current_plan_index_ > body_plan_msg_->plan_indices.back()) {
       ref_body_plan_.row(i) =
           quad_utils::bodyStateMsgToEigen(body_plan_msg_->states.back().body);
+      if (i < N_) {
```

Comment:
There are two if blocks with the same condition. They can be merged I think

Reply:
Not easily, they fall under different conditions and are protecting against out-of-range errors.

Reply:
I realized that after adding the comments, its fine